syntax = "proto3";

package register;

import "common.proto";

service Register {
    rpc register_server(ReachabilityInfo) returns (RegisterResponse);
    rpc get_all_servers_info(common.Empty) returns (AllServerInfo);
}


message ReachabilityInfo {
    string ip_address = 1;
    int32 assignment_port = 2;
    int32 ping_port = 3;
}

message RegisterResponse {
    string server_id = 1;
}

message ServerInfo {
    ReachabilityInfo reachability_info = 1;
    string server_id = 2;
}

message AllServerInfo {
    repeated ServerInfo all_server_info = 1;
}



